pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -486,9 +492,13 @@ func TestChangefeedIdleness(t *testing.T) {
 		cf2 := feed(t, f, "CREATE CHANGEFEED FOR TABLE bar WITH resolved='10ms'")
 		defer closeFeed(t, cf1)
 
+		// Changefeed should run after the insert, but start the check before it
+		// because the changefeed may become idle again before the next
+		// opportunity to check occurs.
```

Comment:
Is it possible that the test may still flake if changefeed becomes non-idle for just one instant, and SucceedsSoon misses that during its time.Sleep? https://github.com/cockroachdb/cockroach/blob/049c30ac6956fa8f307ba2adae8b4388f5907859/pkg/util/retry/retry.go#L220

Aside, this explanation makes sense, but I'm confused why this wasn't flaking before. Longer test latency doesn't seem to check out the flakiness here since it feels like what happened here was the changefeed became non-idle and idle again in a shorter duration? How does it feel to you?